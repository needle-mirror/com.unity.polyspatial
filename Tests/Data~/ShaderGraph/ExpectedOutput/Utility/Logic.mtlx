<?xml version ="1.0"?>
<materialx version = "1.38" colorspace = "lin_rec709">
    <dot name = "_1" type = "float">
        <input name = "in" type = "float" nodename = "_2" />
    </dot>
    <ifequal name = "_2" type = "float">
        <input name = "value1" type = "float" nodename = "_3" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" value = "0" />
        <input name = "in2" type = "float" value = "1" />
    </ifequal>
    <magnitude name = "_3" type = "float">
        <input name = "in" type = "vector3" nodename = "_4" />
    </magnitude>
    <dot name = "_4" type = "vector3">
        <input name = "in" type = "vector3" nodename = "_5" />
    </dot>
    <combine3 name = "_5" type = "vector3">
        <input name = "in1" type = "float" value = "0" />
        <input name = "in2" type = "float" value = "1" />
        <input name = "in3" type = "float" value = "1" />
    </combine3>
    <dot name = "_6" type = "float">
        <input name = "in" type = "float" nodename = "_7" />
    </dot>
    <ifequal name = "_7" type = "float">
        <input name = "value1" type = "float" nodename = "_8" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" value = "0" />
        <input name = "in2" type = "float" value = "1" />
    </ifequal>
    <multiply name = "_8" type = "float">
        <input name = "in1" type = "float" nodename = "_9" />
        <input name = "in2" type = "float" nodename = "_13" />
    </multiply>
    <multiply name = "_9" type = "float">
        <input name = "in1" type = "float" nodename = "_10" />
        <input name = "in2" type = "float" nodename = "_12" />
    </multiply>
    <swizzle name = "_10" type = "float">
        <input name = "in" type = "vector3" nodename = "_11" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <dot name = "_11" type = "vector3">
        <input name = "in" type = "vector3" nodename = "_5" />
    </dot>
    <swizzle name = "_12" type = "float">
        <input name = "in" type = "vector3" nodename = "_11" />
        <input name = "channels" type = "string" value = "y" />
    </swizzle>
    <swizzle name = "_13" type = "float">
        <input name = "in" type = "vector3" nodename = "_11" />
        <input name = "channels" type = "string" value = "z" />
    </swizzle>
    <dot name = "_14" type = "float">
        <input name = "in" type = "float" nodename = "_15" />
    </dot>
    <ifequal name = "_15" type = "float">
        <input name = "value1" type = "float" nodename = "_16" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" value = "0" />
        <input name = "in2" type = "float" value = "1" />
    </ifequal>
    <add name = "_16" type = "float">
        <input name = "in1" type = "float" nodename = "_17" />
        <input name = "in2" type = "float" nodename = "_19" />
    </add>
    <absval name = "_17" type = "float">
        <input name = "in" type = "float" nodename = "_18" />
    </absval>
    <dot name = "_18" type = "float">
        <input name = "in" type = "float" nodename = "_1" />
    </dot>
    <absval name = "_19" type = "float">
        <input name = "in" type = "float" nodename = "_20" />
    </absval>
    <dot name = "_20" type = "float">
        <input name = "in" type = "float" nodename = "_6" />
    </dot>
    <dot name = "_21" type = "float">
        <input name = "in" type = "float" nodename = "_22" />
    </dot>
    <ifequal name = "_22" type = "float">
        <input name = "value1" type = "float" nodename = "_23" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" value = "0" />
        <input name = "in2" type = "float" value = "1" />
    </ifequal>
    <multiply name = "_23" type = "float">
        <input name = "in1" type = "float" nodename = "_24" />
        <input name = "in2" type = "float" nodename = "_25" />
    </multiply>
    <dot name = "_24" type = "float">
        <input name = "in" type = "float" value = "1" />
    </dot>
    <dot name = "_25" type = "float">
        <input name = "in" type = "float" value = "1" />
    </dot>
    <dot name = "_26" type = "float">
        <input name = "in" type = "float" nodename = "_27" />
    </dot>
    <ifequal name = "_27" type = "float">
        <input name = "value1" type = "float" nodename = "_28" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" value = "0" />
        <input name = "in2" type = "float" value = "1" />
    </ifequal>
    <multiply name = "_28" type = "float">
        <input name = "in1" type = "float" nodename = "_29" />
        <input name = "in2" type = "float" nodename = "_31" />
    </multiply>
    <ifequal name = "_29" type = "float">
        <input name = "value1" type = "float" nodename = "_30" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" value = "0" />
    </ifequal>
    <dot name = "_30" type = "float">
        <input name = "in" type = "float" nodename = "_14" />
    </dot>
    <ifequal name = "_31" type = "float">
        <input name = "value1" type = "float" nodename = "_32" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" value = "0" />
    </ifequal>
    <dot name = "_32" type = "float">
        <input name = "in" type = "float" nodename = "_21" />
    </dot>
    <dot name = "_33" type = "float">
        <input name = "in" type = "float" nodename = "_34" />
    </dot>
    <ifequal name = "_34" type = "float">
        <input name = "value1" type = "float" nodename = "_35" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" value = "0" />
    </ifequal>
    <dot name = "_35" type = "float">
        <input name = "in" type = "float" nodename = "_26" />
    </dot>
    <dot name = "_36" type = "float">
        <input name = "in" type = "float" nodename = "_37" />
    </dot>
    <ifequal name = "_37" type = "float">
        <input name = "value1" type = "float" nodename = "_38" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" value = "0" />
    </ifequal>
    <add name = "_38" type = "float">
        <input name = "in1" type = "float" nodename = "_39" />
        <input name = "in2" type = "float" nodename = "_41" />
    </add>
    <realitykit_step name = "_39" type = "float">
        <input name = "in" type = "float" nodename = "_40" />
        <input name = "edge" type = "float" value = "0" />
    </realitykit_step>
    <dot name = "_40" type = "float">
        <input name = "in" type = "float" nodename = "_42" />
    </dot>
    <realitykit_step name = "_41" type = "float">
        <input name = "in" type = "float" value = "0" />
        <input name = "edge" type = "float" nodename = "_40" />
    </realitykit_step>
    <sqrt name = "_42" type = "float">
        <input name = "in" type = "float" nodename = "_Negative" />
    </sqrt>
    <constant name = "_Negative" type = "float">
        <input name = "value" type = "float" value = "-1" />
    </constant>
    <dot name = "_43" type = "float">
        <input name = "in" type = "float" nodename = "_44" />
    </dot>
    <ifequal name = "_44" type = "float">
        <input name = "value1" type = "float" nodename = "_45" />
        <input name = "value2" type = "float" nodename = "_47" />
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" value = "0" />
    </ifequal>
    <absval name = "_45" type = "float">
        <input name = "in" type = "float" nodename = "_46" />
    </absval>
    <dot name = "_46" type = "float">
        <input name = "in" type = "float" nodename = "_48" />
    </dot>
    <divide name = "_47" type = "float">
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" value = "0" />
    </divide>
    <constant name = "_Zero" type = "float">
        <input name = "value" type = "float" value = "0" />
    </constant>
    <divide name = "_48" type = "float">
        <input name = "in1" type = "float" value = "-1" />
        <input name = "in2" type = "float" nodename = "_Zero" />
    </divide>
    <dot name = "_49" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_50" />
    </dot>
    <combine4 name = "_50" type = "vector4">
        <input name = "in1" type = "float" nodename = "_51" />
        <input name = "in2" type = "float" nodename = "_52" />
        <input name = "in3" type = "float" nodename = "_53" />
        <input name = "in4" type = "float" nodename = "_54" />
    </combine4>
    <dot name = "_51" type = "float">
        <input name = "in" type = "float" nodename = "_33" />
    </dot>
    <dot name = "_52" type = "float">
        <input name = "in" type = "float" nodename = "_36" />
    </dot>
    <dot name = "_53" type = "float">
        <input name = "in" type = "float" nodename = "_43" />
    </dot>
    <dot name = "_54" type = "float">
        <input name = "in" type = "float" value = "0" />
    </dot>
    <dot name = "_55" type = "vector3">
        <input name = "in" type = "vector3" nodename = "_56" />
    </dot>
    <combine3 name = "_56" type = "vector3">
        <input name = "in1" type = "float" nodename = "_51" />
        <input name = "in2" type = "float" nodename = "_52" />
        <input name = "in3" type = "float" nodename = "_53" />
    </combine3>
    <dot name = "_57" type = "vector2">
        <input name = "in" type = "vector2" nodename = "_58" />
    </dot>
    <combine2 name = "_58" type = "vector2">
        <input name = "in1" type = "float" nodename = "_51" />
        <input name = "in2" type = "float" nodename = "_52" />
    </combine2>
    <GeometryModification name = "_59" type = "vertexshader">
    </GeometryModification>
    <realitykit_unlit name = "_60" type = "surfaceshader">
        <input name = "applyPostProcessToneMap" type = "boolean" value = "false" />
        <input name = "color" type = "color3" nodename = "_61" />
    </realitykit_unlit>
    <surfacematerial name = "USD_Logic" type = "material">
        <input name = "vertexshader" type = "vertexshader" nodename = "_59" />
        <input name = "surfaceshader" type = "surfaceshader" nodename = "_60" />
    </surfacematerial>
    <swizzle name = "_61" type = "color3">
        <input name = "in" type = "vector3" nodename = "_55" />
        <input name = "channels" type = "string" value = "xyz" />
    </swizzle>
</materialx>
