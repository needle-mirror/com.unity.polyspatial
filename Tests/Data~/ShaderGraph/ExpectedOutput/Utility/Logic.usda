#usda 1.0
(
    metersPerUnit = 1
    upAxis = "Y"
)

def "MaterialX"
{
    def "Materials"
    {
        def Material "USD_Logic"
        {
            float inputs:_Negative = -1
            float inputs:_Zero = 0

            token outputs:mtlx:surface.connect = <realitykit_unlit.outputs:out>
            token outputs:realitykit:vertex.connect = <GeometryModifier.outputs:out>

            def Shader "realitykit_unlit"
            {
                uniform token info:id = "ND_realitykit_unlit_surfaceshader"
                bool inputs:applyPostProcessToneMap = 0
                color3f inputs:color.connect = <../_61.outputs:out>
                token outputs:out
            }

            def Shader "GeometryModifier"
            {
                uniform token info:id = "ND_realitykit_geometrymodifier_vertexshader"
                token outputs:out
            }

            def Shader "_61"
            {
                uniform token info:id = "ND_swizzle_vector3_color3"
                float3 inputs:in.connect = <../_55.outputs:out>
                string inputs:channels = "xyz"
                color3f outputs:out
            }

            def Shader "_55"
            {
                uniform token info:id = "ND_dot_vector3"
                float3 inputs:in.connect = <../_56.outputs:out>
                float3 outputs:out
            }

            def Shader "_56"
            {
                uniform token info:id = "ND_combine3_vector3"
                float inputs:in1.connect = <../_51.outputs:out>
                float inputs:in2.connect = <../_52.outputs:out>
                float inputs:in3.connect = <../_53.outputs:out>
                float3 outputs:out
            }

            def Shader "_51"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_33.outputs:out>
                float outputs:out
            }

            def Shader "_33"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_34.outputs:out>
                float outputs:out
            }

            def Shader "_34"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_35.outputs:out>
                float inputs:value2 = 0
                float inputs:in1 = 1
                float inputs:in2 = 0
                float outputs:out
            }

            def Shader "_35"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_26.outputs:out>
                float outputs:out
            }

            def Shader "_26"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_27.outputs:out>
                float outputs:out
            }

            def Shader "_27"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_28.outputs:out>
                float inputs:value2 = 0
                float inputs:in1 = 0
                float inputs:in2 = 1
                float outputs:out
            }

            def Shader "_28"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_29.outputs:out>
                float inputs:in2.connect = <../_31.outputs:out>
                float outputs:out
            }

            def Shader "_29"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_30.outputs:out>
                float inputs:value2 = 0
                float inputs:in1 = 1
                float inputs:in2 = 0
                float outputs:out
            }

            def Shader "_30"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_14.outputs:out>
                float outputs:out
            }

            def Shader "_14"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_15.outputs:out>
                float outputs:out
            }

            def Shader "_15"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_16.outputs:out>
                float inputs:value2 = 0
                float inputs:in1 = 0
                float inputs:in2 = 1
                float outputs:out
            }

            def Shader "_16"
            {
                uniform token info:id = "ND_add_float"
                float inputs:in1.connect = <../_17.outputs:out>
                float inputs:in2.connect = <../_19.outputs:out>
                float outputs:out
            }

            def Shader "_17"
            {
                uniform token info:id = "ND_absval_float"
                float inputs:in.connect = <../_18.outputs:out>
                float outputs:out
            }

            def Shader "_18"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_1.outputs:out>
                float outputs:out
            }

            def Shader "_1"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_2.outputs:out>
                float outputs:out
            }

            def Shader "_2"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_3.outputs:out>
                float inputs:value2 = 0
                float inputs:in1 = 0
                float inputs:in2 = 1
                float outputs:out
            }

            def Shader "_3"
            {
                uniform token info:id = "ND_magnitude_vector3"
                float3 inputs:in.connect = <../_4.outputs:out>
                float outputs:out
            }

            def Shader "_4"
            {
                uniform token info:id = "ND_dot_vector3"
                float3 inputs:in.connect = <../_5.outputs:out>
                float3 outputs:out
            }

            def Shader "_5"
            {
                uniform token info:id = "ND_combine3_vector3"
                float inputs:in1 = 0
                float inputs:in2 = 1
                float inputs:in3 = 1
                float3 outputs:out
            }

            def Shader "_52"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_36.outputs:out>
                float outputs:out
            }

            def Shader "_36"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_37.outputs:out>
                float outputs:out
            }

            def Shader "_37"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_38.outputs:out>
                float inputs:value2 = 0
                float inputs:in1 = 1
                float inputs:in2 = 0
                float outputs:out
            }

            def Shader "_38"
            {
                uniform token info:id = "ND_add_float"
                float inputs:in1.connect = <../_39.outputs:out>
                float inputs:in2.connect = <../_41.outputs:out>
                float outputs:out
            }

            def Shader "_41"
            {
                uniform token info:id = "ND_realitykit_step_float"
                float inputs:in = 0
                float inputs:edge.connect = <../_40.outputs:out>
                float outputs:out
            }

            def Shader "_40"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_42.outputs:out>
                float outputs:out
            }

            def Shader "_42"
            {
                uniform token info:id = "ND_sqrt_float"
                float inputs:in.connect = <../.inputs:_Negative>
                float outputs:out
            }

            def Shader "_39"
            {
                uniform token info:id = "ND_realitykit_step_float"
                float inputs:in.connect = <../_40.outputs:out>
                float inputs:edge = 0
                float outputs:out
            }

            def Shader "_53"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_43.outputs:out>
                float outputs:out
            }

            def Shader "_43"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_44.outputs:out>
                float outputs:out
            }

            def Shader "_44"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_45.outputs:out>
                float inputs:value2.connect = <../_47.outputs:out>
                float inputs:in1 = 1
                float inputs:in2 = 0
                float outputs:out
            }

            def Shader "_47"
            {
                uniform token info:id = "ND_divide_float"
                float inputs:in1 = 1
                float inputs:in2 = 0
                float outputs:out
            }

            def Shader "_45"
            {
                uniform token info:id = "ND_absval_float"
                float inputs:in.connect = <../_46.outputs:out>
                float outputs:out
            }

            def Shader "_46"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_48.outputs:out>
                float outputs:out
            }

            def Shader "_48"
            {
                uniform token info:id = "ND_divide_float"
                float inputs:in1 = -1
                float inputs:in2.connect = <../.inputs:_Zero>
                float outputs:out
            }

            def Shader "_31"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_32.outputs:out>
                float inputs:value2 = 0
                float inputs:in1 = 1
                float inputs:in2 = 0
                float outputs:out
            }

            def Shader "_32"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_21.outputs:out>
                float outputs:out
            }

            def Shader "_21"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_22.outputs:out>
                float outputs:out
            }

            def Shader "_22"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_23.outputs:out>
                float inputs:value2 = 0
                float inputs:in1 = 0
                float inputs:in2 = 1
                float outputs:out
            }

            def Shader "_23"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_24.outputs:out>
                float inputs:in2.connect = <../_25.outputs:out>
                float outputs:out
            }

            def Shader "_25"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in = 1
                float outputs:out
            }

            def Shader "_24"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in = 1
                float outputs:out
            }

            def Shader "_19"
            {
                uniform token info:id = "ND_absval_float"
                float inputs:in.connect = <../_20.outputs:out>
                float outputs:out
            }

            def Shader "_20"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_6.outputs:out>
                float outputs:out
            }

            def Shader "_6"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_7.outputs:out>
                float outputs:out
            }

            def Shader "_7"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_8.outputs:out>
                float inputs:value2 = 0
                float inputs:in1 = 0
                float inputs:in2 = 1
                float outputs:out
            }

            def Shader "_8"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_9.outputs:out>
                float inputs:in2.connect = <../_13.outputs:out>
                float outputs:out
            }

            def Shader "_9"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_10.outputs:out>
                float inputs:in2.connect = <../_12.outputs:out>
                float outputs:out
            }

            def Shader "_10"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_11.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_11"
            {
                uniform token info:id = "ND_dot_vector3"
                float3 inputs:in.connect = <../_5.outputs:out>
                float3 outputs:out
            }

            def Shader "_13"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_11.outputs:out>
                string inputs:channels = "z"
                float outputs:out
            }

            def Shader "_12"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_11.outputs:out>
                string inputs:channels = "y"
                float outputs:out
            }
        }
    }
}
