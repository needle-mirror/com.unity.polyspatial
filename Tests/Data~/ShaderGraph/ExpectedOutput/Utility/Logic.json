{"properties":[{"name":"_Negative","type":1},{"name":"_Zero","type":1}]}